packages/flutter_tools/lib/src/build_system/file_hash_store.dart

```diff
@@ -148,12 +149,19 @@ class FileHashStore {
     try {
       final String absolutePath = file.path;
       final String previousHash = previousHashes[absolutePath];
-      final Digest digest = md5.convert(await file.readAsBytes());
-      final String currentHash = digest.toString();
-      if (currentHash != previousHash) {
+      // If the file is missing it is assumed to be dirty.
+      if (!file.existsSync()) {
+        currentHashes.remove(absolutePath);
+        previousHashes.remove(absolutePath);
         dirty.add(file);
+      } else {
```

Comment:
Can you return early at the end of the if() {} and remove the else ?

Reply:
Done